controllers/status/status.go

```diff
@@ -210,7 +216,135 @@ func RemoveComponentCondition(conditions *[]conditionsv1.Condition, component st
 }
 
-// ModelRegistryStatus struct holds the status for the ModelRegistry component.
+// +k8s:deepcopy-gen=true
+type ComponentReleaseStatus struct {
+	Name    string                  `json:"name,omitempty"`
+	Version version.OperatorVersion `json:"version,omitempty"`
+	RepoURL string                  `json:"repoURL,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type ComponentStatus struct {
+	Releases []ComponentReleaseStatus `json:"releases,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type CodeFlareStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type DashboardStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type WorkbenchesStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type ModelMeshServingStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type DataSciencePipelinesStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type KserveStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type KueueStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type RayStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type TrustyAIStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
 type ModelRegistryStatus struct {
 	RegistriesNamespace string `json:"registriesNamespace,omitempty"`
+	ComponentStatus     `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type TrainingOperatorStatus struct {
+	ComponentStatus `json:",inline"`
+}
+
+// +k8s:deepcopy-gen=true
+type ComponentsStatus struct {
+	CodeFlare            *CodeFlareStatus            `json:"codeflare,omitempty"`
+	Dashboard            *DashboardStatus            `json:"dashboard,omitempty"`
+	DataSciencePipelines *DataSciencePipelinesStatus `json:"datasciencepipelines,omitempty"`
+	ModelMeshServing     *ModelMeshServingStatus     `json:"modelmeshserving,omitempty"`
+	ModelRegistry        *ModelRegistryStatus        `json:"modelregistry,omitempty"`
+	Kserve               *KserveStatus               `json:"kserve,omitempty"`
+	Kueue                *KueueStatus                `json:"kueue,omitempty"`
+	Ray                  *RayStatus                  `json:"ray,omitempty"`
+	TrustyAI             *TrustyAIStatus             `json:"trustyai,omitempty"`
+	TrainingOperator     *TrainingOperatorStatus     `json:"trainingoperator,omitempty"`
+	Workbenches          *WorkbenchesStatus          `json:"workbenches,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type ReleaseFileMeta struct {
+	Releases []ComponentReleaseStatusMeta `json:"releases,omitempty"`
+}
+
+// +k8s:deepcopy-gen=true
+type ComponentReleaseStatusMeta struct {
+	name    string `yaml:"name,omitempty"`
+	Version string `yaml:"version,omitempty"`
+	RepoURL string `yaml:"repourl,omitempty"`
```

Comment:
```suggestion
	RepoURL string `yaml:"repoURL,omitempty"`
```

at least in the ComponentReleaseStatus on L223, it was changed like this. Could alternatively change to `repository` if the capitalization is confusing.